### ISSUE TITLE
Ordinary users cannot download files!

### ISSUE THREAD

**Issue by ChakChristopher** [45Drives/cockpit-navigator#41]
## Bug Info

**Describe the bug**
Ordinary users cannot download their own files in the home directory, and the download error prompts: no permission or network problems, but root users can download. This problem exists in the Chrome and Firefox browsers of win10 20H2, Ubuntu20.04, centos8.4, but the safari browser of macOS is normal.

**Expected behavior**
Ordinary users can download files in their home directories.

**Screenshots**

![1](https://user-images.githubusercontent.com/37139576/144013220-83500bfd-01d6-4ac7-b874-139d58e5cda8.PNG)

**Comment by joshuaboud** [comment]
Should be fixed with #42, I'll create a release shortly

**Comment by joshuaboud** [comment]
The patch is live in v0.5.8, let me know if the issue is fixed.

**Comment by ChakChristopher** [comment]
Thanks a lot!

**Comment by ChakChristopher** [comment]
The problem has been solved, thank you very much!

**Comment by ChakChristopher** [comment]
Now I found another problem. After installing the update, it can be downloaded after the first login, but it may still fail to download the file after multiple logins. I need to refresh the page and click the download again to succeed. I look forward to your solving this problem. Thank you!

**Comment by joshuaboud** [comment]
Sorry I'm not sure I understand. How can you log in again without the page refreshing? Or do you mean after multiple downloads of the same file?

**Comment by joshuaboud** [comment]
While trying to replicate, I noticed that trying to download an empty file seems to break downloading until refresh. Is this the issue you're experiencing?

**Comment by ChakChristopher** [comment]
If I download a file, it is not empty, and sometimes the download fails, but at this time I click the refresh button in the upper left corner of chrome, and download the file again will succeed.

**Comment by joshuaboud** [comment]
Can you check the output of `journalctl -u cockpit` to see if it's the same as in issue #22? May be a duplicate of this.

**Comment by ChakChristopher** [comment]
Yes, my question is the same as #22.What should I do to solve this problem？

**Comment by joshuaboud** [comment]
You can do what the original poster of #22 did and unblock your server's IP address in Chrome, or a more thorough approach would be creating and installing a certificate for Cockpit. Here is a starting point for that: https://cockpit-project.org/guide/latest/https. It seems that Cockpit should automatically generate a self-signed cert in `/etc/cockpit/ws-certs.d/0-self-signed.cert` that can be copied over and imported into your browser. That page also says it generates a cert named `ca.cert` on some platforms, so if that's there then use that one. Let us know if that works.
